src/platform/packages/private/kbn-index-editor/src/hooks/use_add_column_name.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import useObservable from 'react-use/lib/useObservable';
+import { useCallback, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { KibanaContextExtra } from '../types';
+
+export const useAddColumnName = () => {
+  const {
+    services: { indexUpdateService, notifications },
+  } = useKibana<KibanaContextExtra>();
+
+  const columns = useObservable(indexUpdateService.dataTableColumns$, []);
+
+  const [columnName, setColumnName] = useState('');
+
+  const saveNewColumn = useCallback(async () => {
```

Comment:
This is very weird, I don't like the fact we are returning a boolean here. It should be a () => void function imho.

You can create a separate variable which checks when a field should be added or not
```
const canSave = useMemo(() => columnName && !columns.some((existingColumn) => existingColumn.name === columnName) , [columnName, columns])

```

Reply:
Thank you, much cleaner 👌